src/components/TodoItem.tsx

```diff
@@ -0,0 +1,107 @@
+import { useContext, useState } from 'react';
+import classNames from 'classnames';
+
+import { Todo } from '../types/Todo';
+import { deleteTodo, updateTodos } from '../api/todos';
+import { TodoContext } from '../contexts/TodoContext';
+import { ErrorMessage } from '../types/ErrorMessage';
+import { EditTodoForm } from './EditTodoForm';
+
+interface Props {
+  todo: Todo,
+  isTempTodo?: boolean,
+}
+
+export const TodoItem: React.FC<Props> = ({ todo, isTempTodo }) => {
+  const [isEditMode, setIsEditMode] = useState(false);
+  const {
+    setErrorMessage,
+    setTodos,
+    updateTodoList,
+    idsToUpdate,
+    idsToChange,
+  } = useContext(TodoContext);
+
+  const handleDeleteTodo = (todoId: number) => {
+    idsToUpdate(todoId);
+
+    deleteTodo(todoId)
+      .then(() => setTodos(prevTodos => prevTodos
+        .filter(todoToFilter => todoToFilter.id !== todoId)))
+      .catch(() => setErrorMessage(ErrorMessage.FailedDeleteTodo))
+      .finally(() => {
+        idsToUpdate(null);
+      });
+  };
```

Comment:
it looks like you have the same code 3 times
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/91946036/e4460e07-0c5e-4945-a296-5e7ee4b1471c)
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/91946036/c5762e46-d34a-40ea-9420-246c398f65f3)

could you do some code refactoring?

you can keep this in TodoContext and reuse everywhere you need